jib-core/src/test/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStepTest.java

```diff
@@ -409,4 +418,95 @@ public void testGetCachedBaseImages_v22ManifestListCached_onlyPlatforms()
+    Mockito.verify(eventHandlers).dispatch(LogEvent.debug("mirror config: unmatched2 --> mirror2"));
+    Mockito.verify(buildContext, Mockito.never()).newBaseImageRegistryClientFactory(Mockito.any());
+  }
+
+  @Test
+  public void testTryMirrors_mirrorIoError()
+      throws LayerCountMismatchException, BadContainerConfigurationFormatException, IOException,
+          RegistryException {
+    Mockito.when(imageConfiguration.getImageRegistry()).thenReturn("registry");
+    Mockito.when(buildContext.getRegistryMirrors())
+        .thenReturn(ImmutableListMultimap.of("registry", "gcr.io"));
+    Mockito.when(buildContext.newBaseImageRegistryClientFactory("gcr.io"))
+        .thenReturn(registryClientFactory);
+    Mockito.when(registryClient.pullManifest(Mockito.any()))
+        .thenThrow(new IOException("test exception"));
+
+    Optional<ImagesAndRegistryClient> result =
+        pullBaseImageStep.tryMirrors(buildContext, progressDispatcherFactory);
+    Assert.assertEquals(Optional.empty(), result);
+
+    Mockito.verify(eventHandlers)
+        .dispatch(LogEvent.info("trying mirror gcr.io for the base image"));
+    Mockito.verify(eventHandlers)
+        .dispatch(LogEvent.debug("failed to get manifest from mirror gcr.io: test exception"));
+  }
+
+  @Test
+  public void testTryMirrors_multipleMirrors()
+      throws LayerCountMismatchException, BadContainerConfigurationFormatException, IOException,
+          RegistryException, InvalidImageReferenceException {
+    Mockito.when(imageConfiguration.getImage()).thenReturn(ImageReference.parse("registry/repo"));
+    Mockito.when(imageConfiguration.getImageRegistry()).thenReturn("registry");
+    Mockito.when(buildContext.getRegistryMirrors())
+        .thenReturn(ImmutableListMultimap.of("registry", "quay.io", "registry", "gcr.io"));
+
+    Mockito.when(buildContext.newBaseImageRegistryClientFactory("quay.io"))
+        .thenReturn(registryClientFactory);
+    Mockito.when(registryClient.pullManifest(Mockito.any()))
+        .thenThrow(new RegistryException("not found"));
+
+    RegistryClient.Factory gcrRegistryClientFactory = setUpWorkingRegistryClientFactory();
+    Mockito.when(buildContext.newBaseImageRegistryClientFactory("gcr.io"))
+        .thenReturn(gcrRegistryClientFactory);
+
+    Optional<ImagesAndRegistryClient> result =
+        pullBaseImageStep.tryMirrors(buildContext, progressDispatcherFactory);
+    Assert.assertTrue(result.isPresent());
+    Assert.assertEquals(gcrRegistryClientFactory.newRegistryClient(), result.get().registryClient);
+
+    Mockito.verify(eventHandlers)
+        .dispatch(LogEvent.info("trying mirror quay.io for the base image"));
+    Mockito.verify(eventHandlers)
+        .dispatch(LogEvent.info("trying mirror gcr.io for the base image"));
+    Mockito.verify(eventHandlers)
+        .dispatch(LogEvent.debug("failed to get manifest from mirror quay.io: not found"));
```

Comment:
if this is the expected log output ordering, I think it's a little confusing?

I would expect it to be
```
trying quay..
failed to get from quay...
trying gcr..
```

Reply:
Used `Mockito.InOrder()` and rearranged the order.